lib/handlers/deliverTxHandlerFactory.js

```diff
@@ -0,0 +1,87 @@
+const cbor = require('cbor');
+const { Transaction } = require('@dashevo/dashcore-lib');
+const {
+  ApplyStateTransitionRequest,
+} = require('@dashevo/drive-grpc');
+const {
+  abci: {
+    ResponseDeliverTx,
+  },
+} = require('abci/types');
+
+const InvalidArgumentAbciError = require('./errors/InvalidArgumentAbciError');
+
+/**
+ * deliverTx ABCI handler
+ * @param {DashPlatformProtocol} dpp
+ * @param {AppState} state
+ * @param {UpdateStatePromiseClient} updateStateClient
+ * @returns {deliverTxHandler}
+ */
+module.exports = function deliverTxHandlerFactory(dpp, state, updateStateClient) {
+  /**
+   * @typedef deliverTxHandler
+   * @param request
+   * @returns {Promise<abci.ResponseDeliverTx>}
+   */
+  async function deliverTxHandler(request) {
+    const tx = Buffer.from(request.tx, 'base64');
```

Comment:
I think we need to encapsulate this logic in a function and call it here and in `checkTx`.